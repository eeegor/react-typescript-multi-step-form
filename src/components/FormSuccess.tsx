/**
 * Component FormSuccess
 */

import * as React from 'react';
import * as classnames from 'classnames';
import './FormSuccess.scss';
import { IconSuccess } from './IconSuccess';
import { Button } from './Button';

export interface Props {
	className?: string;
	onReset?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export class FormSuccess extends React.Component<Props> {
	render(): JSX.Element {
		const { className, onReset } = this.props;

		return (
			<div className={classnames('form-success', className)}>
				<IconSuccess id="success" color="#83c736" />
				<h2>Success!</h2>
				<p>Thank you for submitting</p>
				<Button type="success" onClick={event => onReset && onReset(event)}>
					Reset form
				</Button>
			</div>
		);
	}
}
