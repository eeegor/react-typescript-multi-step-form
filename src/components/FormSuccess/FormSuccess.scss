.form-success {
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	padding: 0;
	display: flex;
	align-items: center;
	overflow-y: scroll;
	text-align: center;

	&__container {
		margin: auto;
		padding: 4rem 0 8rem 0;
		width: 100%;
	}
}

.form-success__reset {
	margin: 0 auto;
}
