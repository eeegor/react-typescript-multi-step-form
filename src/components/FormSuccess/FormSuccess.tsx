/**
 * Component FormSuccess
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { IconSuccess, Button } from '..';
import './FormSuccess.scss';

export interface Props {
	className?: string;
	onReset?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export class FormSuccess extends React.Component<Props> {
	render(): JSX.Element {
		const { className, onReset } = this.props;

		return (
			<div className={classnames('form-success', className)}>
				<IconSuccess id="success" color="#83c736" />
				<h2>Success!</h2>
				<p>Thank you for submitting</p>
				<Button className="form-success__reset" type="primary" onClick={event => onReset && onReset(event)}>
					Start over
				</Button>
			</div>
		);
	}
}
