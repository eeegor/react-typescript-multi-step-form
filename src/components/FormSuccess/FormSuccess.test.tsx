import React from 'react';
import renderer from 'react-test-renderer';
import { FormSuccess } from './FormSuccess';
import { mount } from 'enzyme';

describe('FormSuccess', () => {
	const callback = jest.fn();

	it('renders markup correctly', () => {
		const wrapper = <FormSuccess onReset={callback} />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('click event works', () => {
		const wrapper = mount(<FormSuccess onReset={callback} />);
		wrapper
			.find('.form-success__reset')
			.first()
			.simulate('click');
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
