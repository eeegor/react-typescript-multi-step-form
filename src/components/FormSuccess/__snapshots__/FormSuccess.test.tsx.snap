// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSuccess renders markup correctly 1`] = `
<div
  className="form-success"
>
  <div
    className="form-success__container"
  >
    <div
      className="icon-success"
    >
      <svg
        height="160px"
        viewBox="0 0 72 72"
        width="160px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          stroke="#8EC343"
          strokeWidth="3"
        >
          <circle
            className="icon-success__circle"
            cx="36"
            cy="36"
            r="34"
            style={
              Object {
                "strokeDasharray": "240px, 240px",
                "strokeDashoffset": "480px",
              }
            }
          />
          <path
            className="icon-success__path"
            d="M17.417,37.778l9.93,9.909l25.444-25.393"
            style={
              Object {
                "strokeDasharray": "50px, 50px",
                "strokeDashoffset": "0px",
              }
            }
          />
        </g>
      </svg>
    </div>
    <h2>
      Success!
    </h2>
    <p>
      Thank you for submitting
    </p>
    <button
      className="button button--primary form-success__reset"
      onClick={[Function]}
      type="button"
    >
      Start over
    </button>
  </div>
</div>
`;
